tests/integration_tests/functional/test_topology.py

```diff
@@ -78,35 +78,75 @@ def _check_cache_topology_x86(
     )
 
 
-def _check_cache_topology_arm(test_microvm, no_cpus):
+def _aarch64_parse_cache_info(test_microvm, no_cpus):
+    def parse_cache_info(info: str):
+        "One line looks like this: /sys/devices/system/cpu/cpuX/cache/{index}/{name}:{value}"
+        cache_info = []
+        for line in info.splitlines():
+            parts = line.split("/")
+
+            index = int(parts[-2][-1])
+
+            data = parts[-1].split(":")
+            name = data[0]
+            value = data[1]
+
+            if len(cache_info) == index:
+                cache_info.append({})
+            cache_info[index][name] = value
+        return cache_info
+
     # We will check the cache topology by looking at what each cpu
     # contains as far as cache info.
     # For that we are iterating through the hierarchy of folders inside:
     # /sys/devices/system/cpu/cpuX/cache/indexY/type - the type of the cache
     # (i.e Instruction, Data, Unified)
     # /sys/devices/system/cpu/cpuX/cache/indexY/size - size of the cache
     # /sys/devices/system/cpu/cpuX/cache/indexY/level - L1, L2 or L3 cache.
-    # There are 2 types of L1 cache (instruction and data) that is why the
-    # "cache_info" variable below has 4 items.
-
-    sys_cpu = "/sys/devices/system/cpu"
-    fields = ["level", "type", "size", "coherency_line_size", "number_of_sets"]
-
-    cmd = f"grep . {sys_cpu}/cpu{{0..{no_cpus-1}}}/cache/index*/{{{','.join(fields)}}} |sort"
+    fields = ["level", "type", "size"]
+    cmd = f"grep . /sys/devices/system/cpu/cpu{{0..{no_cpus-1}}}/cache/index*/{{{','.join(fields)}}} |sort"
 
     _, guest_stdout, guest_stderr = test_microvm.ssh.run(cmd)
     assert guest_stderr == ""
 
-    res = subprocess.run(
+    host_result = subprocess.run(
         cmd,
         shell=True,
         executable="/bin/bash",
         capture_output=True,
         check=True,
         encoding="ascii",
     )
-    assert res.stderr == ""
-    assert res.stdout == guest_stdout
+    assert host_result.stderr == ""
+    host_stdout = host_result.stdout
+
+    guest_cache_info = parse_cache_info(guest_stdout)
+    host_cache_info = parse_cache_info(host_stdout)
+
+    return guest_cache_info, host_cache_info
+
+
+def _check_cache_topology_arm(test_microvm, no_cpus, kernel_version):
+    guest_cache_info, host_cache_info = _aarch64_parse_cache_info(test_microvm, no_cpus)
+
+    # Starting from 6.3 kernel cache representation for aarch64 platform has changed.
+    # It is no longer equivalent to the host cache representation.
+    # The main change is in the level 1 cache, so for newer kernels we
+    # compare only level 2 and level 3 caches
+    if kernel_version < "6.3":
```

Comment:
This won't work for kernel `6.10`. Use the tuple `global_props.host_linux_version_tpl < (6, 3)`

Reply:
fixed